service/storage_service.cc

```diff
@@ -1676,12 +1698,15 @@ class topology_coordinator {
                 if (_topo_sm._topology.transition_nodes.empty()) {
                     builder.del_transition_state();
                 } else {
-                    builder.set_transition_state(topology::transition_state::write_both_read_old);
+                    builder.set_transition_state(topology::transition_state::tablet_draining);
                 }
                 auto str = ::format("published CDC generation, ID: {}", curr_gen_id);
                 co_await update_topology_state(std::move(guard), {builder.build()}, std::move(str));
             }
                 break;
+            case topology::transition_state::tablet_draining:
+                co_await handle_tablet_migration(std::move(guard), true);
```

Comment:
Having a separate state means moving out tablets and streaming vnode data will be serialized. But there is no harm doing it in parallel, no?

Reply:
Currently they cannot proceed in parallel because they rely on common infrastructure for barriers and fencing. We cannot have two active coordinators which make topology changes and barriers concurrently, as one will ruin the other. For example, barrier executed by tablets can put up a fence for vnodes too soon resulting in request failures. 

We would have to synchronize them somehow, then we could do them in parallel. But it's not that simple. 

And from practical perspective there may be not much point to do this, because you either have most data in tablets or most data in vnodes, not both, in which case there's not much gain from parallelising them.

Reply:
May be we should have separate topology versions for vnodes and tablets. But if your goal to basically stop using vnodes when tables are available then you last point makes sense.

Reply:
Yes, no need to optimize mixed vnodes/tablets.